src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const curentDateArr = date.split(fromFormat[3]);
+  const newFormatArr = Array(3);
+
+  const fromIndexD = fromFormat.indexOf('DD');
+  const fromIndexM = fromFormat.indexOf('MM');
+  const fromIndexY = 3 - fromIndexD - fromIndexM;
+
+  for (let i = 0; i < 3; i++) {
+    if (toFormat[i].includes('D')) {
+      newFormatArr[i] = curentDateArr[fromIndexD];
+    }
+
+    if (toFormat[i].includes('M')) {
+      newFormatArr[i] = curentDateArr[fromIndexM];
+    }
+
+    if (toFormat[i].includes('Y')) {
+      newFormatArr[i] = curentDateArr[fromIndexY];
```

Comment:
These are the same actions just based on different values and you repeat yourself three times. Try to create object like:
```
{
  D: value,
  M: value,
  Y: value,
}
```
so you can get the first character in `toFormat[i]` and find needed element by this character from object by default